lib/kubeclient/common.rb

```diff
@@ -254,6 +254,10 @@ def define_entity_methods
         do |name, patch, namespace = nil|
           patch_entity(entity.resource_name, name, patch, 'merge-patch', namespace)
         end
+
+        define_singleton_method("apply_#{entity.method_names[0]}") do |*args|
+          apply_entity(entity.resource_name, *args)
```

Comment:
Should this use `**opts` or `*args, **opts` for Ruby 2.7+ compatibility?
https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/#delegation
actually suggests not using `**opts` in code that has to run on Ruby 2.6 or earlier.  But the motivation seems to be a corner case with empty opts that won't happen here if field_manager: is mandatory?  My head hurts thinking about this...

Would it work to define the block with the actual params we expect?
```suggestion
        define_singleton_method("apply_#{entity.method_names[0]}") \
        do |resource, field_manager:, force: true|
          apply_entity(entity.resource_name, resource, field_manager: field_manager, force: force)
```
Not critical, not sure it's better, just thinking aloud...
Current code passed CI, including 2.7, so it's also good.

Reply:
> actually suggests not using `**opts` in code that has to run on Ruby 2.6 or earlier. But the motivation seems to be a corner case with empty opts that won't happen here if field_manager: is mandatory? My head hurts thinking about this...

Indeed using `**opts` breaks:

```
ArgumentError: wrong number of arguments (given 1, expected 0)
    /Users/blangfeld/code/kubeclient/lib/kubeclient/common.rb:258:in `block (2 levels) in define_entity_methods'
    /Users/blangfeld/code/kubeclient/lib/kubeclient/common.rb:101:in `method_missing'
```

If we want true forward compatibility with 2.7/3, we need the ruby2_keywords gem, but I'm not sure if that works with procs or just methods.

> Would it work to define the block with the actual params we expect?

I did that, but it [was suggested to do otherwise](https://github.com/abonas/kubeclient/pull/448#discussion_r441097839). I'll do whatever you're willing to merge.